models/survey.py

```diff
@@ -156,26 +156,19 @@ def get_results(self, survey_id):
     def get_formatted_results(self, survey_id):
         results = self.get_results(survey_id)
 
-        def get_pie_data(question_data, question):
+        def get_data(question_data, question):            
             return r.branch(
-                (r.expr(question_data['response_format'] == Question().RESPONSE_TRUE_OR_FALSE)),
-                question[1].group(lambda r: r).count().ungroup().map(
-                    lambda gr: {
-                        'name': gr['group'].coerce_to('string'),
-                        'value': gr['reduction']
-                    }
-                ),
-                []
-            )
-
-        def get_bar_data(question_data, question):
-            return r.branch(
-                (r.expr(question_data['response_format'] == Question().RESPONSE_MULTIPLE_CHOICE) | (question_data['response_format'] == Question().RESPONSE_RATING)),
+                (r.expr(question_data['response_format'] == Question().RESPONSE_MULTIPLE_CHOICE) | (question_data['response_format'] == Question().RESPONSE_RATING) | (question_data['response_format'] == Question().RESPONSE_TRUE_OR_FALSE)),
                 r.branch(
                     (question_data['response_format'] == Question().RESPONSE_MULTIPLE_CHOICE),
                     {
-                        'labels': question[1].distinct(),
-                        'series': [question[1].distinct().map(lambda val: question[1].filter(lambda foo: foo == val).count())]
+                        'labels': question_data['options'],
+                        'series': [r.expr(question[1]).reduce(lambda left, right: left.map(right, lambda leftVal, rightVal: leftVal + rightVal))]
```

Comment:
If they do I can just as well just put it in an array on the front end, bar graphs just require an array of arrays on the front end or they fail =/